spec/models/user_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+RSpec.describe User, :type => :model do
+  subject do
+    @user1 = FactoryBot.create(:user)
+  end
+
+  describe '#valid attributes' do
+    it 'is valid with valid attributes' do
+      subject
+      expect(@user1).to be_valid
+    end
+  end
+
+  describe '#unique mail, (uid,provider)' do
```

Comment:
Maybe for this uniqueness and presence tests you could use the `shoulda-matchers` ([link](https://github.com/thoughtbot/shoulda-matchers)). With this gem you can check the presence and uniqueness validations easier and in one line. You can use: 

```
it { is_expected.to validate_uniqueness_of(:email) }
it { is_expected.to validate_presence_of(:email) }
```

For this to work correctly you have to add some config to the `rails_helper.rb` file ([link](https://github.com/thoughtbot/shoulda-matchers#rails-apps)).
Also [here](https://github.com/thoughtbot/shoulda-matchers#matchers) you have a list of the available matchers to use.